### PR TITLE
#1905 created checks in each cr endpoints to check if it's reviewed

### LINKED ISSUE #1905: Auto Implementing CRs - Add Checks to all Create New CR Endpoints
### Description

We need to validate that the wbs element that is requesting a change only has one unreviewed change request up at a time to avoid merge conflicts. 

### Acceptance Criteria

- [ ] Add a check that throws an http error if the wbs element associated with the change request has any unreviewed change requests open. 

### Proposed Solution

^

### Mocks

epic #1892

### PR DESCRIPTION
**Pull request by AShyamkumar19**
## Changes

Added checks to the 3 create change request methods. Checks to see if it's reviewed

## Test Cases

none

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1905

### CODE REVIEW

File: src/backend/src/services/change-requests.services.ts [hunk 1]
```diff
@@ -384,6 +383,14 @@ export default class ChangeRequestsService {
       );
     }
 
+    const reviewedProject = wbsElement.changeRequests.filter((cr) => cr.dateReviewed === null);
```
Comment (Aaryan1203) [inline src/backend/src/services/change-requests.services.ts:386]: make this reviewedProjects with an s because this is technically multiple projects

Comment (Aaryan1203) [reply src/backend/src/services/change-requests.services.ts:386]: also it should be unreviewedChangeRequests

File: src/backend/src/services/change-requests.services.ts [hunk 2]
```diff
@@ -514,6 +530,14 @@ export default class ChangeRequestsService {
       });
     }
 
+    const reviewedProject = wbsElement.changeRequests.filter((cr) => cr.dateReviewed === null);
```
Comment (Aaryan1203) [inline src/backend/src/services/change-requests.services.ts:533]: same thing here

File: src/backend/src/services/change-requests.services.ts [hunk 3]
```diff
@@ -456,7 +463,16 @@ export default class ChangeRequestsService {
           workPackageNumber
         }
       },
-      include: { workPackage: { include: { expectedActivities: true, deliverables: true } }, changeRequests: true }
+      include: {
+        workPackage: {
+          include: {
+            expectedActivities: true,
+            deliverables: true,
+            project: { include: { teams: true, wbsElement: true } }
```
Comment (Aaryan1203) [inline src/backend/src/services/change-requests.services.ts:471]: I don't think you need this line

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
left a couple of comments but great work!

**Review by Aaryan1203 — APPROVED** [review]
Amazing 🫡

**Comment by Aaryan1203** [comment]
Also assign yourself to the pr

**Comment by RChandler234** [comment]
These tickets should've been in blocked, we still need finalize some details, that's my b

**Pull request closed without merging**